prime-router/src/main/kotlin/fhirengine/engine/FHIREngine.kt

```diff
@@ -0,0 +1,222 @@
+            } else {
+                settingsProvider ?: settingsProviderSingleton
+            }
+
+            return FHIREngine(
+                metadata ?: Metadata.getInstance(),
+                settingsProvider!!,
+                databaseAccess ?: databaseAccessSingleton,
+                blobAccess ?: BlobAccess(),
+                queueAccess ?: QueueAccess
+            )
+        }
+    }
+
+    /**
+     * Process a [message] off of the raw-elr azure queue, convert it into FHIR, and store for next step.
+     * [workflowEngine] is used for some underlying functionality that should be pulled out into a base class
+     * at some point (tech debt).
+     * [actionHistory] and [actionLogger] ensure all activities are logged
+     */
+    fun processHL7(
+        message: RawSubmission,
+        actionLogger: ActionLogger,
+        actionHistory: ActionHistory,
+        hl7Reader: HL7Reader = HL7Reader(actionLogger)
+    ) {
+        logger.trace("Processing HL7 data for FHIR conversion.")
+        try {
+            // get the hl7 from the blob store
+            val hl7messages = hl7Reader.getMessages(message.downloadContent())
+
+            if (actionLogger.hasErrors()) {
+                throw java.lang.IllegalArgumentException(actionLogger.errors.joinToString("\n") { it.detail.message })
+            }
+
+            // use fhir transcoder to turn hl7 into FHIR
+            val fhirBundles = hl7messages.map {
+                HL7toFhirTranslator.getInstance().translate(it)
+            }
+
+            logger.info("Received ${fhirBundles.size} FHIR bundles.")
+
+            actionHistory.trackExistingInputReport(message.reportId)
+
+            // operate on each fhir bundle
+            for (bundle in fhirBundles) {
+                // make a 'report'
+                val report = Report(
+                    Report.Format.FHIR,
+                    emptyList(),
+                    fhirBundles.size,
+                    itemLineage = listOf(
+                        ItemLineage()
+                    )
+                )
+
+                // create item lineage
+                report.itemLineages = listOf(
+                    ItemLineage(
+                        null,
+                        message.reportId,
+                        1,
+                        report.id,
+                        1,
+                        null,
+                        null,
+                        null,
+                        report.getItemHashForRow(1)
+                    )
+                )
+
+                // create translate event
+                val translateEvent = ProcessEvent(
+                    Event.EventAction.TRANSLATE,
+                    report.id,
+                    Options.None,
+                    emptyMap<String, String>(),
+                    emptyList<String>()
+                )
+
+                // upload to blobstore
+                var bodyBytes = FhirTranscoder.encode(bundle).toByteArray()
```

Comment:
Just a note that I think does not matter: The JSON in the file is not pretty printed, so at some point I can see someone asking for this to be formatted, so it is easier to read.  I did this in the ProcessFHIRCommand CLI class.